app/src/main/java/ch/epfl/sdp/peakar/utils/SettingsUtilities.java

```diff
@@ -0,0 +1,77 @@
+package ch.epfl.sdp.peakar.utils;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.util.DisplayMetrics;
+
+import java.util.Locale;
+
+import ch.epfl.sdp.peakar.R;
+
+/**
+ *  Utility class for the Settings Activity, contains utility methods for the settings like the language
+ */
+public final class SettingsUtilities {
+
+    private static final String DEFAULT_LANGUAGE = "en";
+
+    /**
+     * Updates the language depending on the preferences
+     *
+     * @param context context of the application
+     * @param sharedPreferences shared preferences
+     */
+    public static void updateLanguage(Context context, SharedPreferences sharedPreferences){
+        String language =  sharedPreferences.getString(context.getResources().getString(R.string.language_key), DEFAULT_LANGUAGE);
+        SettingsUtilities.setLocale(context, SettingsUtilities.getLanguageCode(language));
+    }
+
+    /**
+     * Get language code from language
+     *
+     * @param language language
+     * @return language code (2 letters)
+     */
+    private static String getLanguageCode(String language){
+        switch (language){
+            case "german":
+                return "de";
+            case "italian":
+                return "it";
+            case "french":
+                return "fr";
+            case "swedish":
+                return "sv";
+            case "english":
```

Comment:
Why isn't there any return here?

Reply:
case "english" and and the default return the same value so I made the case "english" fall through. Basically, if the case is "english" then the flow of control will fall to the next case which is the default case that returns "en". I could also simply delete the case "english" but I think that it's better to display the case.